# PART 1: SALES AND PROFITABILITY ANALYSIS

In [1]:
# Importing necessary library
import pandas as pd

In [2]:
# Creating a dataframe
df = pd.read_csv("/content/drive/MyDrive/Colab Notebooks/Merged_orders.csv")
df

Order ID  Amount  Profit  Quantity     Category      Sub-Category  \
0     B-25601    1275   -1148         7    Furniture         Bookcases   
1     B-25601      66     -12         5     Clothing             Stole   
2     B-25601       8      -2         3     Clothing       Hankerchief   
3     B-25601      80     -56         4  Electronics  Electronic Games   
4     B-25602     168    -111         2  Electronics            Phones   
...       ...     ...     ...       ...          ...               ...   
1495  B-26099     835     267         5  Electronics            Phones   
1496  B-26099    2366     552         5     Clothing          Trousers   
1497  B-26100     828     230         2    Furniture            Chairs   
1498  B-26100      34      10         2     Clothing           T-shirt   
1499  B-26100      72      16         2     Clothing             Shirt   

      Order Date Customer Name           State       City  
0     01-04-2018        Bharat         Gujarat  Ahmedabad  
1     01-04-2018        Bharat         Gujarat  Ahmedabad  
2     01-04-2018        Bharat         Gujarat  Ahmedabad  
3     01-04-2018        Bharat         Gujarat  Ahmedabad  
4     01-04-2018         Pearl     Maharashtra       Pune  
...          ...           ...             ...        ...  
1495  30-03-2019        Bhishm     Maharashtra     Mumbai  
1496  30-03-2019        Bhishm     Maharashtra     Mumbai  
1497  31-03-2019        Hitika  Madhya Pradesh     Indore  
1498  31-03-2019        Hitika  Madhya Pradesh     Indore  
1499  31-03-2019        Hitika  Madhya Pradesh     Indore  

[1500 rows x 10 columns]

In [19]:
# Checks how many categories are present
df['Category'].unique()

array(['Furniture', 'Clothing', 'Electronics'], dtype=object)

In [4]:
profits = df.groupby('Category')

In [5]:
category_grouped = df.groupby("Category")

In [6]:
# Calculating total profit, amount and orders for each category
total_profit = category_grouped['Profit'].sum()
total_amount = category_grouped['Amount'].sum()
total_orders = category_grouped["Order ID"].nunique()

In [7]:
#average profit for each category
avg_profit = round(total_profit/total_orders,2)
avg_profit

Category
Clothing       28.40
Electronics    51.44
Furniture      12.35
dtype: float64

These are the average profit per order for each category.

In [8]:
# Calculating Profit margin for each category
profit_margin = round((total_profit/total_amount)*100,2)
profit_margin

Category
Clothing       8.03
Electronics    6.35
Furniture      1.81
dtype: float64

In [9]:
# Merging all result to present clear view
result = pd.DataFrame({"Average Profit per Order": avg_profit,"Total Profit Margin(%)": profit_margin}).reset_index()
result

Category  Average Profit per Order  Total Profit Margin(%)
0     Clothing                     28.40                    8.03
1  Electronics                     51.44                    6.35
2    Furniture                     12.35                    1.81

These are the average profit per order and total profit margin for each category and the profit margin values are in percentage.

In [10]:
count_orders = sum(total_orders)
count_orders

783

In [11]:
total_orders

Category
Clothing       393
Electronics    204
Furniture      186
Name: Order ID, dtype: int64

In [12]:
pos_profit = df[df["Profit"]>0]
neg_profit = df[df["Profit"]<0]

In [13]:
# Counting Positive profit order and sum of the profits
pos_stats = pos_profit.groupby("Category")["Profit"].agg(["count","sum"]).reset_index()
pos_stats

Category  count    sum
0     Clothing    631  23156
1  Electronics    180  27216
2    Furniture    139  17819

In [14]:
# Counting Negative profit order and sum of the profits
neg_stats = neg_profit.groupby("Category")["Profit"].agg(["count","sum"]).reset_index()
neg_stats

Category  count    sum
0     Clothing    279 -11993
1  Electronics    122 -16722
2    Furniture    102 -15521

In [15]:
total_orders_per_cat = df.groupby("Category")["Profit"].count().astype(float)

In [16]:
# Merging all the results
profit_percent = pd.merge(pos_stats,neg_stats, on="Category",how="outer").fillna(0)
profit_percent["Total Orders"] = profit_percent["Category"].map(total_orders_per_cat)
profit_percent

Category  count_x  sum_x  count_y  sum_y  Total Orders
0     Clothing      631  23156      279 -11993         949.0
1  Electronics      180  27216      122 -16722         308.0
2    Furniture      139  17819      102 -15521         243.0

In [17]:
# Calculating positive and negative percentage for each category
profit_percent["Positive %"]=(profit_percent["count_x"]/profit_percent["Total Orders"])*100
profit_percent["Negative %"]=(profit_percent["count_y"]/profit_percent["Total Orders"])*100
profit_percent = profit_percent.drop(columns=["Total Orders"])
profit_percent

Category  count_x  sum_x  count_y  sum_y  Positive %  Negative %
0     Clothing      631  23156      279 -11993   66.491043   29.399368
1  Electronics      180  27216      122 -16722   58.441558   39.610390
2    Furniture      139  17819      102 -15521   57.201646   41.975309

In [18]:
# Renaming column name for understanding.
profit_percent.rename(columns={'count_x':'Positive Orders','sum_x':'Positive Profit','count_y':'Negative Orders','sum_y':'Negative Profit'},inplace= True)
profit_percent

Category  Positive Orders  Positive Profit  Negative Orders  \
0     Clothing              631            23156              279   
1  Electronics              180            27216              122   
2    Furniture              139            17819              102   

   Negative Profit  Positive %  Negative %  
0           -11993   66.491043   29.399368  
1           -16722   58.441558   39.610390  
2           -15521   57.201646   41.975309

In these, the top-performing category is `Clothing` which has 631 profitable orders with a `8.03%` profit margin and the low-performing category is `Furniture` which has only 139 profitable orders with a `1.81%` profit margin.

* Here we can see that clothing has 2/3 of positive profit where only 1/4 of negative profit which can be manageble and In furniture 3/5 positive profit but 5/12 of negative profit which is almost 50/50 chances.

* You can also note that there is only 241 orders were placed where 102 orders gave negative profit but clothing holds a total of 631 successful orders out of 910.

* Focus on product pricing and plan strategies to attract more customers